### PR TITLE
Add `TypedData.toJsonString()`

### LINKED ISSUE #481: Add `TypedData.toJsonString()` method
### Feature Request

Self-explanatory, the need for adding this feature comes from one of community members.

### PR DESCRIPTION
**Pull request by franciszekjob**
## Describe your changes
- Add `TypedData.toJsonString()` method
- Add `TypeDataTest.JsonConversionTest`

## Linked issues

Closes #481 

## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

File: lib/src/test/kotlin/starknet/data/TypedDataTest.kt [hunk 1]
```diff
@@ -704,6 +706,73 @@ internal class TypedDataTest {
...
+            domain = domainObject,
+            message = messageObject,
+        )
+        private val tdJsonString by lazy { File("$TYPE_DATA_DIR_PATH/rev_1/typed_data_basic_types_example.json").readText() }
+
+        @Test
+        fun `typed data toJsonString`() {
+            // TypedData.toJsonString() returns a JSON string without indentations
+            // So we need to compare it with the expected JSON string without indentations
+            val expectedJsonString = Json.encodeToString(Json.parseToJsonElement(tdJsonString))
```
Comment (franciszekjob) [inline lib/src/test/kotlin/starknet/data/TypedDataTest.kt:764]: `TypedData.toJsonString()` returns a minified JSON string, so we need to convert pretty-printed `tdJsonString` to minified format.

File: lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt [hunk 2]
```diff
@@ -601,6 +605,14 @@ data class TypedData private constructor(
         fun fromJsonString(typedData: String): TypedData =
             Json.decodeFromString(serializer(), typedData)
     }
+
+    /**
+     * Create a JSON string from TypedData.
+     *
+     * @param typedData typed data instance
```
Comment (ddoktorski) [inline lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt:612]: Technically, it doesn't accept any parameters.
```suggestion
```

File: lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt [hunk 3]
```diff
@@ -601,6 +605,14 @@ data class TypedData private constructor(
...
+     */
+    internal fun toJsonString(): String =
```
Comment (ddoktorski) [inline lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt:614]: Could you explain why this is an internal function?

File: lib/src/test/kotlin/starknet/data/TypedDataTest.kt [hunk 4]
```diff
@@ -22,8 +22,10 @@ import org.junit.jupiter.params.provider.MethodSource
 import java.io.File
 import java.math.BigInteger
 
+private const val TYPE_DATA_DIR_PATH = "src/test/resources/typed_data"
```
Comment (ddoktorski) [inline lib/src/test/kotlin/starknet/data/TypedDataTest.kt:25]: nit:
```suggestion
private const val TYPED_DATA_DIR_PATH = "src/test/resources/typed_data"
```

File: lib/src/test/kotlin/starknet/data/TypedDataTest.kt [hunk 5]
```diff
@@ -704,6 +706,72 @@ internal class TypedDataTest {
...
+            ),
+        )
+        private val domainObject = """
+            {
+                "name": "StarkNet Mail",
+                "version": "1",
+                "chainId": "1",
+                "revision": 1
+            }
+        """.trimIndent()
```
Comment (ddoktorski) [inline lib/src/test/kotlin/starknet/data/TypedDataTest.kt:738]: Is `trimIndent()` necessary here? Why isn't it used in the `messageObject` below?

Comment (franciszekjob) [reply lib/src/test/kotlin/starknet/data/TypedDataTest.kt:738]: > Is trimIndent() necessary here?

`trimIndent()` is used in every JSON string in `TypedDataTest` and in other tests in our SDK - that's why I wanted to follow this "standard".

> Why isn't it used in the messageObject below?

Didn't notice it - added 🫡

File: lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt (hunk 3 above)
Comment (franciszekjob) [reply lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt:613]: Actually no need to keep it internal, as we want to allow users to use it.

File: lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt [hunk 6]
```diff
@@ -601,6 +605,13 @@ data class TypedData private constructor(
         fun fromJsonString(typedData: String): TypedData =
             Json.decodeFromString(serializer(), typedData)
     }
+
+    /**
+     * Create a JSON string from TypedData.
+     *
```
Comment (ddoktorski) [inline lib/src/main/kotlin/com/swmansion/starknet/data/TypedData.kt:611]: nit: This line can be removed as well
```suggestion
```

**Review by ddoktorski — APPROVED** [review]

**Pull request merged**